src/main/java/org/example/repository/UserRepositoryImpl.java

```diff
@@ -0,0 +1,187 @@
+package org.example.repository;
+
+import org.example.model.User;
+import org.example.repository.dao.UserRepository;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserRepositoryImpl implements UserRepository {
+    private final Connection connection;
+    private static final String selectAllById =
+            """
+                            SELECT *
+                            FROM users
+                            WHERE id = ?
+                    """;
+
+    private static final String register =
+            """
+                            INSERT INTO users(
+                            name, birthday, amount, nickname, password)
+                            VALUES (?, ?, ?, ?, ?)
+                    """;
+
+    private static final String recharge =
+            """
+                            UPDATE users
+                            SET amount = ?
+                            WHERE  id = ?;
+                    """;
+
+    private static final String userGames =
+            """
+                            SELECT users_games.games_id 
+                            FROM users_games
+                            WHERE users_games.users_id = ?
+                    """;
+
+    private static final String addGameToUser =
+            """
+                    INSERT INTO "users_games"(users_id, games_id)
+                    VALUES (?, ?);
+                    """;
```

Comment:
all of these queries use only one time, so it's better to move them in appropriate methods